tutorials/intro-express/1-welcome.qmd

```diff
@@ -0,0 +1,310 @@
+the text (output component) will react and update to the corresponding input value.
+
+* Inputs are created via `ui.input_*()` functions.
+    * The first argument is the input's `id`, which is used to read the input's value.
+* Outputs are created by decorating a function with `@render.*`.
+    * Inside a `render` function, `input` values can be read [reactively](#reactivity).
+    * When those `input` values change, Shiny knows how to minimally re-render output.
+* Layouts are inferred automatically based on what items you place in your application.
+    * We will learn more about layouts and user interfaces in the next lesson of this tutorial.
+
+::: {.callout-note}
+## Exercise
+
+Let's make and run our first shiny for python application.
+
+1. Take the above code and save it to a file. Here we named it `app.py`
+2. Click on the play button (red circle in the image below)j
+
+You will see the terminal run the `shiny run` command for you automatically.
+The output will look something like this
+
+
+```bash
+$ python -m shiny run --port 55901 --reload --autoreload-port 55902 app-010-simple.py
+INFO:     Will watch for changes in these directories: ['~/Desktop/py-shiny-example']
+INFO:     Uvicorn running on http://127.0.0.1:55901 (Press CTRL+C to quit)
+INFO:     Started reloader process [24969] using WatchFiles
+INFO:     Started server process [24986]
+INFO:     Waiting for application startup.
+INFO:     Application startup complete.
+INFO:     127.0.0.1:56426 - "GET /?vscodeBrowserReqId=1737097751843 HTTP/1.1" 200 OK
+```
+
+This will run the application on port `55901` and automatically reload and update
+as you make changes to the `app.py` file.
+
+1. You will see the app build in the bottom terminal and open in the viewer on the side
+2. Move the slider and see how the output reacts
+3. Congratulations, you made your first shiny for python application!
+
+![](img/010-run_app.png)
+
+::: {.callout-tip}
+## Naming your files
+
+If you start your file with the word `app`,
+the shiny for python extension will recognize
+it as an application and you will be able to see the "play" button to run your application.
+You can also name your file `app-getting_started.py`
+and you will still get the shiny extension play button.
+
+To have Shiny for Python work well with the VS Code extensions and for you to go through
+the next series of lessons.
+We recommend either one of the following file naming conventions:
+
+1. Create separate folders for each app example you will create and save separate `app.py` files in each folder
+2. Create separate `app*.py` files in the same directory (e.g., `app-01.py`, `app-02.py`)
+:::
+
+:::
+
+## Run your shiny application
+
+In addition to the play button in Positron, you can manually run your application from
+the command line.
+This is useful if you wish to specify your own port or want to rename your application
+without the `app` prefix.
+
+```bash
+shiny run my_app.py
+```
+
+::: {.callout-note}
```

Comment:
I generally prefer that we give callouts informative titles, it makes them easier to read in the prose and helps them stand out from each other.